com.collins.trustedsystems.jkindapi/src/jkind/lustre/visitors/AstIterVisitor.java

```diff
@@ -32,13 +76,59 @@ public Void visit(Function e) {
 		return null;
 	}
 
+	@Override
+	public Void visit(Guarantee e) {
+		e.expr.accept(this);
+		return null;
+	}
+
+	@Override
+	public Void visit(ImportedFunction e) {
+		visitVarDecls(e.inputs);
+		visitVarDecls(e.outputs);
+		// if (e.contractBody != null) {
+		//	visit(e.contractBody);
+		// }
+		return null;
+	}
+
+	@Override
+	public Void visit(ImportedNode e) {
+		visitVarDecls(e.inputs);
+		visitVarDecls(e.outputs);
+		// if (e.contractBody != null) {
+		//	visit(e.contractBody);
+		// }
+		return null;
+	}
+
+	@Override
+	public Void visit(Kind2Function e) {
+		visitVarDecls(e.inputs);
+		visitVarDecls(e.outputs);
+		visitVarDecls(e.locals);
+		visitEquations(e.equations);
+		visitAssertions(e.assertions);
+		return null;
+	}
+
+	@Override
+	public Void visit(Mode e) {
+		visitExprs(e.require);
+		visitExprs(e.ensure);
+		return null;
+	}
+
 	@Override
 	public Void visit(Node e) {
 		visitVarDecls(e.inputs);
 		visitVarDecls(e.outputs);
 		visitVarDecls(e.locals);
 		visitEquations(e.equations);
 		visitAssertions(e.assertions);
+		// if (e.contractBody != null) {
+		//	visit(e.contractBody);
+		// }
```

Comment:
The original code for traversing the node skipped the inline contract. I followed suit and commented out code for traversing Kind 2 constructs in this file and `AstMapVisitor.java`. Please let me know if that's the intended behavior.